packages/app/src/LogSidePanel.tsx

```diff
@@ -1342,10 +1336,149 @@ function PropertySubpanel({
+      lineWrap: true,
+      useLegacyViewer: false,
+    },
+  );
+
+  const getLineActions = useCallback<GetLineActions>(
+    ({ keyPath, value }) => {
+      const actions: LineAction[] = [];
+
+      if (onPropertyAddClick != null && typeof value !== 'object') {
+        actions.push({
+          key: 'add-to-search',
+          label: <i className="bi bi-plus-circle" />,
+          title: 'Add to Search',
+          onClick: () => {
+            onPropertyAddClick(`${keyPath.join('.')}`, value);
+          },
+        });
+      }
+
+      if (typeof value !== 'object') {
+        actions.push({
+          key: 'search',
+          label: <i className="bi bi-search" />,
+          title: 'Search for this value only',
+          onClick: () => {
+            router.push(
+              generateSearchUrl(
+                `${keyPath.join('.')}:${
+                  typeof value === 'string' ? `"${value}"` : value
+                }`,
+              ),
+            );
+          },
+        });
+      }
+
+      /* TODO: Handle bools properly (they show up as number...) */
+      if (typeof value === 'number') {
+        actions.push({
+          key: 'chart',
+          label: <i className="bi bi-graph-up" />,
+          title: 'Chart',
+          onClick: () => {
+            router.push(
+              generateChartUrl({
+                aggFn: 'avg',
+                field: `${keyPath.join('.')}`,
+                groupBy: [],
+                table: 'logs',
+              }),
+            );
+          },
+        });
+      }
+
+      if (toggleColumn && typeof value !== 'object') {
+        const keyPathString = keyPath.join('.');
+        actions.push({
+          key: 'toggle-column',
+          label: <i className="bi bi-table" />,
+          title: displayedColumns?.includes(keyPathString)
+            ? `Remove ${keyPathString} column from results table`
+            : `Add ${keyPathString} column to results table`,
+          onClick: () => toggleColumn(keyPathString),
+        });
+      }
+
+      const handleCopyObject = () => {
+        const shouldCopyParent = !isNestedView;
+        const parentKeyPath = keyPath.slice(0, -1);
+        const copiedObj = shouldCopyParent
+          ? parentKeyPath.length === 0
+            ? nestedProperties
+            : get(nestedProperties, parentKeyPath)
+          : keyPath.length === 0
+          ? nestedProperties
+          : get(nestedProperties, keyPath);
+        window.navigator.clipboard.writeText(
+          JSON.stringify(copiedObj, null, 2),
+        );
+        toast.success(
+          `Copied ${shouldCopyParent ? 'parent' : 'object'} to clipboard`,
+        );
+      };
+
+      if (typeof value === 'object') {
+        actions.push(
+          isNestedView
+            ? {
+                key: 'copy-object',
+                label: 'Copy Object',
+                onClick: handleCopyObject,
+              }
+            : {
+                key: 'copy-parent',
```

Comment:
It seems like there's no way to actually ask to copy a parent any more in the flat view. On the old viewer this was accomplishable by clicking the key of the object, in the flat view.

Iiirc the use case was that users woud like to find a specific key/value via searching the flat view, and then copy the parent object for further use. I think we can alternatively allow users to search the nested view and copy objects there which might be cleaner.